### PR TITLE
Fix Multi-Change Event Errors

### PR DESCRIPTION
**Pull request by BrChung**
### Description

Fixing a bug addressed in issue #380   

An on-change event with multiple operations would throw an error. Examples of these operations include: `Ctrl + Z`, `Text Drag`, `Multi-line comment`, etc.

Fix:
The bug is fixed by ensuring that the offset is added to restLength. Previously, the offset is added then subtracted, causing a negative value to be passed into TextOperation.retain()

The operations provided by Monaco are in-order; therefore, reversing the changes would cause errors.

### CODE REVIEW

**Comment by samtstern** [comment]
@BrChung thank you for sending this!  Would you mind rebasing on top of `master` (or just merge `master` in to your branch).  I recently updated the CI to stop failing so we'll be able to ensure tests pass on this PR.

File: lib/monaco-adapter.js [hunk 1]
```diff
@@ -373,8 +373,8 @@ var MonacoAdapter = function () {
         this.trigger('change', op, op);
       }
 
-      /** Reverse iterate all changes */
-      event.changes.reverse().forEach(function (change) {
+      /** Iterate through all changes */
+      event.changes.forEach(function (change) {
```
Comment (samtstern) [inline lib/monaco-adapter.js:377]: Full disclosure: I maintain this library but was not involved at all in writing it.  I have a feeling there was a good reason to use `reverse()` in the first place.  Do you understand what the reason for that processing was and why it's ok to change it now?

Comment (BrChung) [reply lib/monaco-adapter.js:377]: I've had similar thoughts while reviewing the code. I do not fully understand the use of `reverse().` If I were to guess, perhaps it was to address the way that Monaco emitted changes in a previous version.

Based on my debugging, reversing the changes would consistently throw an error.

Example (Using the same steps from example 1 from issue #380):
When undoing multiple changes after deleting the last two characters from the word "Hello."

The correct result should be
Hel --*Ctrl + Z*-> Hel + l (offset: 3) + o (offset: 4)

Monaco seems to return the changes in the correct order; reversing the changes would result with:
Hel --*Ctrl + Z*-> Hel + o (offset: 4) + l (offset: 3)

The inconsistency with the offset then throws an error further down the operation chain.
Screen Shot 2020-09-14 at 9 19 36 AM

The same applies to other multi-change events I've tested, such as multi-line commenting (Ctrl + ?), find/replace all occurrences, text-dragging, etc.

File: lib/monaco-adapter.js [hunk 2]
```diff
@@ -304,7 +304,7 @@ var MonacoAdapter = function () {
     /** Change Informations */
     var text = change.text;
     var rangeLength = change.rangeLength;
-    var rangeOffset = change.rangeOffset + offset;
```
Comment (BrChung) [inline lib/monaco-adapter.js:307]: Removing offset is causing errors in the Monaco Adapter Unit Tests

**Comment by samtstern** [comment]
@BrChung ok the tests pass and it seems to work in your examples so I think I'm gonna do a little 🤞  and merge this one!  Thank you for the contribution!

**Comment by samtstern** [comment]
This change was released in version `1.5.11`

**Pull request merged**